extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -46,18 +46,20 @@ export class SchemaCompareDialog {
 	private targetIsDacpac: boolean;
 	private database: string;
 	public dialogName: string;
+	private connection: azdata.connection.ConnectionProfile;
 
 	protected initializeDialog(): void {
 		this.schemaCompareTab = azdata.window.createTab(SchemaCompareLabel);
 		this.initializeSchemaCompareTab();
 		this.dialog.content = [this.schemaCompareTab];
 	}
 
-	public openDialog(p: any, dialogName?: string): void {
+	public async openDialog(p: any, dialogName?: string): Promise<void> {
```

Comment:
Why not have p be IConnectionProfile?

Reply:
isn't it ObjectExplorerContext not IConnectionProfile?